### PR TITLE
MailTo Transport Test Cases

### PR DESCRIPTION
**Pull request by irhamucsc**

### CODE REVIEW

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+     * @throws Exception - Error while sending the gmail
+     */
+    public static boolean sendMailAndCheckReceived(String emailSubject) throws Exception {
+        Properties props = getSMTPProperties();
+        EmailSender emailSender =
+                new EmailSender(props, sender, String.valueOf(senderPassword), domain, receiver + "@" + domain);
+        if (emailSender.createSession()) {
+            emailSender.setSubject(emailSubject);
+            emailSender.setBody("Body : " + emailSubject);
+            emailSender.sendEmail();
```
Comment (krishanthasamaraweera) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: sendEmail throws Exception and we need to fix it in next framework release to send custom or specific exception properly.

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+                try {
+                    store.close();
+                } catch (MessagingException e) {
+                    log.error("Error when closing the store ", e);
+                }
+            }
+        }
+    }
+
+    /**
```
Comment (krishanthasamaraweera) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Method description is missing in the doc comment

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+     * @return boolean - If found the email true , else false
+     * @throws InterruptedException - Error occurred in thread sleep
+     * @throws ESBMailTransportIntegrationTestException   - Is thrown if an error while reading the emails
+     */
+    public static boolean waitToCheckEmailReceived(String subjectToCheck)
+            throws ESBMailTransportIntegrationTestException {
+        boolean mailReceived = false;
+        long startTime = System.currentTimeMillis();
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (isMailReceivedBySubject(subjectToCheck, INBOX)) {
```
Comment (krishanthasamaraweera) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Rename the "INBOX" constant as "EMAIL_INBOX"

File: modules/integration/tests-integration/tests-transport/src/test/java/org/wso2/carbon/esb/mailto/transport/sender/test/SendMailWithBCCThroughESBTestCase.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        loadESBConfigurationFromClasspath(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "mailTransport" +
+                File.separator + "mailTransportSender" + File.separator + "smtpBcc" + File.separator +
+                "mail_sender_bcc.xml");
+
+        MailToTransportUtil.readXMLforEmailCredentials();
+
+        MailToTransportUtil.deleteAllUnreadEmailsFromGmail();
+        // Since ESB reads all unread emails one by one, we have to delete the all unread emails to run the test
```
Comment (krishanthasamaraweera) [inline modules/integration/tests-integration/tests-transport/src/test/java/org/wso2/carbon/esb/mailto/transport/sender/test/SendMailWithBCCThroughESBTestCase.java]: move the comment before the code statement.

File: modules/integration/pom.xml [hunk 5]
```diff
@@ -41,11 +41,11 @@
     </licenses>
 
     <modules>
-        <!--module>test-common</module>
```
Comment (krishanthasamaraweera) [inline modules/integration/pom.xml]: Please check with Product team and enable selected modules. AFAIK, they have disabled certain modules to fix bugs.

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+            return emailReceived;
+        } catch (MessagingException ex) {
+            log.error("Error when getting mail count ", ex);
+            throw new ESBMailTransportIntegrationTestException("Error when getting mail count ", ex);
+        } finally {
+            if (store != null) {
+                try {
+                    store.close();
+                } catch (MessagingException e) {
+                    log.error("Error when closing the store ", e);
```
Comment (krishanthasamaraweera) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Change the log.error to log.warn

File: pom.xml [hunk 7]
```diff
@@ -37,11 +37,11 @@
         <version>1</version>
     </parent>
     <modules>
-        <module>modules/styles</module>
-        <module>modules/samples</module>
-        <module>modules/features</module>
-        <module>modules/p2-profile</module>
-        <module>modules/distribution</module>
+        <!--<module>modules/styles</module>-->
```
Comment (krishanthasamaraweera) [inline pom.xml]: Please review .. Are you commenting the other modules.

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java (hunk 2 above)
Comment (irham0019) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Done

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java (hunk 3 above)
Comment (irham0019) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Done

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java (hunk 6 above)
Comment (irham0019) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Done

File: pom.xml (hunk 7 above)
Comment (irham0019) [inline pom.xml]: Removed comments

File: modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+            for (Message message : messages) {
+                if (message.getSubject().contains(emailSubject)) {
+                    log.info("Found the email emailSubject : " + emailSubject);
+                    emailReceived = true;
+                    break;
+                }
+            }
+            return emailReceived;
+        } catch (MessagingException ex) {
+            log.error("Error when getting mail count ", ex);
```
Comment (sagara-gunathunga) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: Is this the only place you log this exception ? Make sure you don't log same exception in two places.

Comment (irham0019) [inline modules/integration/test-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MailToTransportUtil.java]: I was logging the getConnection() method's exception in two other places . Now i have fixed it,

**Comment by sdkottegoda** [comment]
Closing since this is invalid

**Pull request closed without merging**